File: Python/1239. Maximum Length of a Concatenated String with Unique Charachters.py
class Solution:
    def maxLength(self, arr: List[str]) -> int:
        res = [""]  # Creating a list with empty string
        maxLength = 0

        for i in arr:
            for j in res:
                maxRes = i + j
                if(len(maxRes) != len(set(maxRes))):
                    continue
                # Continue : Current iteration is skipped and we move to execute further st.
                # Break: Current iteration is skipped and we do not execute further st.
                res.append(maxRes)
                maxLength = max(maxLength, len(maxRes))
        return maxLength

s = Solution()
arr = ["un","iq","ue"]
maxL = s.maxLength(arr)
print(maxL)

File: C++/1239. Maximum Length of a Concatenated String with Unique Charachters.cpp
#include <vector>
#include <string>
#include <unordered_set>
#include <algorithm>

class Solution {
public:
    int maxLength(std::vector<std::string>& arr) {
        std::vector<std::string> res {""};
        int maxLength = 0;

        for (auto i : arr) {
            for (auto j : res) {
                std::string maxRes = i + j;
                if (maxRes.size() != std::unordered_set<char>(maxRes.begin(), maxRes.end()).size()) {
                    continue;
                }
                res.push_back(maxRes);
                maxLength = std::max(maxLength, (int)maxRes.size());
            }
        }
        return maxLength;
    }
};

int main() {
    Solution s;
    std::vector<std::string> arr {"un","iq","ue"};
    int maxL = s.maxLength(arr);
    std::cout << maxL << std::endl;
    return 0;
}
